### ISSUE TITLE
be smarter about finding the pair

### ISSUE THREAD

**Issue by jeremychone** [BriteSnow/vscode-toggle-quotes#8]
## Problem

```
"This stri][ng won't work because of the apostrophe."
```

Won't work (from @evaera)

## Solution

- Needs to be smarter to find the quote pair.

**Comment by ryanhaney** [comment]
I think this might be another example: ```"You must specify `options.type`"```

**Comment by weidmaster** [comment]
And yet another example, using Laravel blade syntax: ```"{{ __('Enter Percentage') }}"```

- If you put the cursor near the first quote, it will convert the first apostrophe to a quote and won't change the first quote at all. (```"{{ __("Enter Percentage') }}"```)
- If you put the cursor betweent the apostrophes it will convert both to quotes (CORRECT)(```"{{ __("Enter Percentage") }}"```)
- if you put the cursor near the last quote, it will convert the quotes to apostrophes (CORRECT)(```'{{ __('Enter Percentage') }}'```)

So in this example it is getting wrong the first situation. Maybe there is some way to check if a quote has a pair (like if checking for parenthesis pairs or something).

**Comment by jeremychone** [comment]
For now, you select the area you want to ignore. 

So, you like this "{{ **][**('Enter Percentage')**][** }}" and this will toggle the quotes outside. 

A little overhead from the user point of view, but it allows the plugin to be very predictable (and dumb).

**Comment by weidmaster** [comment]
Yes selecting the area to ignore does work @jeremychone 
I do use the toggle a lot without selecting though because it saves time. But thanks for the workaround.

**Comment by spazmodius** [comment]
This is a decent work-around. It would be good if it was discoverable without spelunking github issues.

**Comment by weidmaster** [comment]
> This is a decent work-around. It would be good if it was discoverable without spelunking github issues.

Oh I suggest @jeremychone to mention this workaround in the plugin overview page inside VSCode. That would help 😄

**Comment by jeremychone** [comment]
@weidmaster YES, I agree, this is long overdue. Will try to do an update by next week. I was thinking about a simple animated gif.

Side note: This is actually the by design way, of "ignore what is selected" this way, give full control (dumb is the new smart). There is a good bigger pull request to add the full ignore escaped quotes, but probably needs to be in a separate keyboard shortcut.
